internal/build/job/signer.go

```diff
@@ -0,0 +1,184 @@
+
+	if pushImage {
+		args = []string{"-signedimage", containerImage}
+	} else {
+		args = append(args, "-no-push")
+	}
+
+	if previousImage != "" {
+		args = append(args, "-unsignedimage", previousImage)
+	} else if buildConfig.Sign.UnsignedImage != "" {
+		args = append(args, "-unsignedimage", buildConfig.Sign.UnsignedImage)
+	} else {
+		return nil, fmt.Errorf("no image to sign given")
+	}
+	args = append(args, "-key", "/signingkey/key.priv")
+	args = append(args, "-cert", "/signingcert/public.der")
+
+	if len(buildConfig.Sign.FileList) > 0 {
+		args = append(args, "-filestosign", strings.Join(buildConfig.Sign.FileList, ":"))
+	}
+
+	fmt.Printf("buildConfig == %#v\n", buildConfig)
+	fmt.Printf("buildConfig.Sign == %#v\n", buildConfig.Sign)
+
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	volumes = append(volumes, m.makeImageSigningSecretVolume(buildConfig.Sign.KeySecret, "key", "key.priv"))
+	volumes = append(volumes, m.makeImageSigningSecretVolume(buildConfig.Sign.CertSecret, "cert", "public.der"))
+
+	volumeMounts = append(volumeMounts, m.makeImageSigningSecretVolumeMount(buildConfig.Sign.CertSecret, "/signingcert"))
+	volumeMounts = append(volumeMounts, m.makeImageSigningSecretVolumeMount(buildConfig.Sign.KeySecret, "/signingkey"))
+
+	if mod.Spec.ImageRepoSecret != nil {
+		args = append(args, "-pullsecret", "/docker_config/config.json")
+		volumes = append(volumes, m.makeImagePullSecretVolume(mod.Spec.ImageRepoSecret))
+		volumeMounts = append(volumeMounts, m.makeImagePullSecretVolumeMount(mod.Spec.ImageRepoSecret))
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: mod.Name + "-" + m.GetName() + "-",
+			Namespace:    mod.Namespace,
+			Labels:       labels(mod, targetKernel, m.GetName()),
+		},
+		Spec: batchv1.JobSpec{
+			Completions: pointer.Int32(1),
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:         "signimage",
+							Image:        "quay.io/chrisp262/kmod-signer:latest",
+							Args:         args,
+							VolumeMounts: volumeMounts,
+						},
+					},
+					RestartPolicy: v1.RestartPolicyOnFailure,
+					Volumes:       volumes,
+					NodeSelector:  mod.Spec.Selector,
+				},
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(&mod, job, m.scheme); err != nil {
+		return nil, fmt.Errorf("could not set the owner reference: %v", err)
+	}
+
+	return job, nil
+}
+
+func (m *signer) makeImageSigningSecretVolume(secretRef *v1.LocalObjectReference, key string, path string) v1.Volume {
+
+	return v1.Volume{
+		Name: m.volumeNameFromSecretRef(*secretRef),
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: secretRef.Name,
+				Items: []v1.KeyToPath{
+					{
+						Key:  key,
+						Path: path,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (m *signer) makeImageSigningSecretVolumeMount(secretRef *v1.LocalObjectReference, mountpoint string) v1.VolumeMount {
```

Comment:
we have makeVolume functions both in signer.go and in maker.go. Can we probably move those functions into some kind of helper file, and uses them with appropriate parameters from both files?